pkg/provider/github/acl.go

```diff
@@ -103,6 +111,26 @@ func (v *Provider) aclCheckAll(ctx context.Context, rev *info.Event) (bool, erro
 	return acl.UserInOwnerFile(ownerContent, rev.Sender)
 }
 
+// checkPullRequestForSameURL checks If PullRequests are for same clone URL and different branches
+// means if the user has access to create a branch in the repository without forking or having any permissions then PAC should allow to run CI.
+//
+//	ex: dependabot, *[bot] etc...
+func (v *Provider) checkPullRequestForSameURL(ctx context.Context, runevent *info.Event) (bool, error) {
+	pr, resp, err := v.Client.PullRequests.Get(ctx, runevent.Organization, runevent.Repository, runevent.PullRequestNumber)
+	if err != nil {
+		return false, err
+	}
+	if resp != nil && resp.StatusCode == http.StatusNotFound {
```

Comment:
don't think a note found can ever happen but yeah no worries to keep it in there